### PR TITLE
Handle non-zero return code from nvidia-smi in check_nvidia()

### PR DESCRIPTION
**Pull request by olliewalsh**
Fixes: #2197

## Summary by Sourcery

Handle failures of the nvidia-smi command in GPU detection and add tests for these scenarios.

Bug Fixes:
- Treat non-zero return codes and execution errors from nvidia-smi as absence of NVIDIA GPUs instead of raising errors.

Tests:
- Add unit tests covering successful nvidia-smi execution, non-zero return code errors, and missing nvidia-smi binary for check_nvidia().

### CODE REVIEW

File: test/unit/test_common.py [hunk 1]
```diff
@@ -170,6 +171,28 @@ def test_apple_vm_returns_result(mock_handle_provider, mock_run_cmd, mock_config
...
+
+    @patch("subprocess.run")
+    def test_check_nvidia_smi_failure(self, mock_run):
+        mock_run.side_effect = subprocess.CalledProcessError(1, "nvidia-smi")
+        assert check_nvidia() is None
+
+    @patch("subprocess.run")
+    def test_check_nvidia_smi_not_found(self, mock_run):
+        mock_run.side_effect = OSError("nvidia-smi not found")
+        assert check_nvidia() is None
```
[the remark this answers is not in the document]
Comment (rhatdan) [reply test/unit/test_common.py:193]: Seems reasonable.

**Comment by rhatdan** [comment]
LGTM
